src/huggingface_hub/inference/_providers/hf_inference.py

```diff
@@ -38,7 +38,13 @@ def _prepare_url(self, api_key: str, mapped_model: str) -> str:
         # hf-inference provider can handle URLs (e.g. Inference Endpoints or TGI deployment)
         if mapped_model.startswith(("http://", "https://")):
             return mapped_model
-        return f"{self.base_url}/models/{mapped_model}"
+        return (
+            # Feature-extraction and sentence-similarity are the only cases where we handle models with several tasks.
+            f"{self.base_url}/pipeline/{self.task}/{mapped_model}"
```

Comment:
note: this URL is probably not super easy to "route" to a specific Pod on the Infra side, but it'd be painful to change it now